packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -574,4 +574,32 @@ describe('<DataGrid /> - Filter', () => {
     };
     render(<Test />);
   });
+
+  it('should apply the valueParser onto the filter value', () => {
+    render(
+      <TestCase
+        rows={[
+          {
+            id: 1,
+            amount: 0.5,
+          },
+          {
+            id: 2,
+            amount: 1,
+          },
+        ]}
+        columns={[
+          {
+            field: 'amount',
+            type: 'number',
+            valueParser: (value) => (value as number) / 100,
+          },
+        ]}
+        operatorValue={'='}
+        value={50}
+        field={'amount'}
```

Comment:
if we want to follow the existing convention:

```suggestion
        operatorValue="="
        value={50}
        field="amount"
```